File: jobs/2026-05-30__07-57-35/multi_service_api_go__cWcV5rF/artifacts/user/myproject/order/order.go
package order

import (
	"context"

	"myapp/user"
)

type Order struct {
	ID       string `json:"id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

// Get returns an order by ID and fetches user details.
//encore:api public method=GET path=/order/:id
func Get(ctx context.Context, id string) (*Order, error) {
	userID := "user-" + id
	userDetails, err := user.Get(ctx, userID)
	if err != nil {
		return nil, err
	}

	return &Order{
		ID:       id,
		UserID:   userDetails.ID,
		UserName: userDetails.Name,
	}, nil
}

File: jobs/2026-07-02__17-09-42/encore_multi_service_api_ts__9Sfh7Yk/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

export interface OrderParams {
  id: number;
}

export interface OrderResponse {
  orderId: number;
  userId: number;
  userName: string;
}

// Returns an order by ID. Always fetches user ID 1 from the user service.
export const getOrder = api<OrderParams, OrderResponse>(
  { expose: true, method: "GET", path: "/order/:id" },
  async ({ id }) => {
    const u = await user.getUser({ id: 1 });
    return {
      orderId: id,
      userId: u.id,
      userName: u.name,
    };
  }
);